src/test/groovy/unit/org/pih/warehouse/api/ProductApiControllerSpec.groovy

```diff
@@ -0,0 +1,121 @@
+package unit.org.pih.warehouse.api
+
+import grails.orm.PagedResultList
+import grails.testing.gorm.DataTest
+import grails.testing.web.controllers.ControllerUnitTest
+import java.util.stream.IntStream
+import org.grails.plugins.testing.GrailsMockHttpServletResponse
+import org.grails.web.json.JSONElement
+import org.grails.web.json.JSONObject
+import org.hibernate.Criteria
+import org.json.JSONArray
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import org.pih.warehouse.api.ProductApiController
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.forecasting.ForecastingService
+import org.pih.warehouse.product.Product
+import org.pih.warehouse.product.ProductService
+
+@Unroll
+class ProductApiControllerSpec extends Specification implements DataTest, ControllerUnitTest<ProductApiController> {
+
+	@Shared private def forecastingServiceStub
+	@Shared private def productServiceStub
+
+    void setupSpec() {
+		mockDomains(Product, Location)
+    }
+
+	void setup() {
+		productServiceStub = Stub(ProductService)
+		forecastingServiceStub = Stub(ForecastingService)
+
+		controller.forecastingService = forecastingServiceStub
+		controller.productService = productServiceStub
```

Comment:
the indentation seems to be too big - we usually use 2 for tab.

Reply:
I think for groovy files we have 4 (at least that's what I have `tab size: 4, indent: 4, Continuation indent 8`), 2 for tab we have for js